### PR TITLE
Cobalt Message

### PR DESCRIPTION
**Pull request by AmalieT**

### CODE REVIEW

File: matscholar_web/search/logic.py [hunk 1]
```diff
@@ -115,3 +118,16 @@ def search_bar_live_display(example_search_n_clicks, *ent_txts):
...
+
+
+def is_cobalt_search(entity_query, raw_text):
+    if raw_text is not None:
+        if 'Co' in raw_text.split() or 'cobalt' in raw_text.lower().split():
+            return True
+
+    if entity_query is not None:
+        if 'material' in entity_query.keys():
+            if any(['Co' in m for m in entity_query['material']]) or any(['cobalt' in m.lower() for m in entity_query['material']]):
```
Comment (ardunn) [inline matscholar_web/search/logic.py]:
```
any(['Co' in m for m in entity_query['material']])..
```

Won't this throw a cobalt warning for things like "Colloidal", if it's entered as a material?

This line might work better as a more precise regex or something
@AmalieT

**Comment by ardunn** [comment]
Besides comment above, looks ok to me!

File: matscholar_web/search/logic.py (hunk 1 above)
Comment (AmalieT) [reply matscholar_web/search/logic.py]: Good point. Changed to use a regex as suggested.

**Comment by ardunn** [comment]
Hey @AmalieT there seem to be some problems with this I didn't consider at first

## 1. There's no tests for this feature. 
Tests are a requirement now for any feature (especially something simple and self contained enough as checking whether there is Cobalt in a material). The two things that should be tested are `is_cobalt_search` and `cobalt_warning_html`. 

#### Testing `cobalt_warning_html`
This should just check whether an html obj is returned; this is important because our app is big and complex now, and making sure functions that say they return html objs actually return html objs is important (otherwise we are opening ourselves up to debugging tons of React nonsense). It would go in `matscholar_web.search.test_common`

I set up some base test class methods to make this [_fingers crossed_] a lot easier - it should require like 3-4 SLOC. If the view function has no arguments, `MatScholarWebBaseTest.run_test_for_all_functions_in_module` will automatically test all view functions in search common view since it is called in `matscholar_web.search.test_common.TestSearchCommonViews` - AKA, you don't have to do anything for it to be tested. However, since `cobalt_warning_html` has at least one arg, just add it to the excluded list (`EXCLUDE`) and write a test for it using the base class method `MatScholarWebBaseTest.run_test_for_individual_arg_combo`. This will call your html view function with a bunch of arguments (or combinations of arguments) you define to make sure it doesn't error out or return bogus types on any of them. `TestSearchCommonView.test_results_label_html` gives an example of how to use it. The args you could test `cobalt_warning_html` with, for example, might be just an empty div in the simplest case. You can just copy + paste `test_results_label_html` and change the function name and arguments in `arg_combos` and it should work, provided `cobalt_warning_html` does in fact work with the args given.

#### Testing `is_cobalt_search`
I was writing some tests for this myself but a few things I'm not quite getting:

A simple case:
```
from matscholar_web.search.logic import is_cobalt_search
eq1 = {"material": "CoMnO2"}
rt1 = "material: CoMnO2"
print(is_cobalt_search(eq1, rt1))
```

Returns False?


Similarly,
```
from matscholar_web.search.logic import is_cobalt_search
eq1 = {"material": "Cobalt", "application": "Stuff"}
rt1 = "material: Cobalt, application: Stuff"
print(is_cobalt_search(eq1, rt1))
```

Also returns False?
I think something in the regex has gone wrong, or I'm not using it right


## 2. `is_cobalt_search` needs docs. 

## 3. Run `source scripts/format_project.sh` to format

**Comment by ardunn** [comment]
Also paging @jdagdelen as a heads up for future features **needing** tests, the first part of (1) above might be of use

**Pull request merged**